### DISCUSSION TITLE
Workflow input through Rest API

### DISCUSSION THREAD

**Discussion by Mishtir** [Netflix/conductor#3085] [Ideas]
Hi,
can we pass workflow input through an rest api in conductor? 
So I have a requirement to pass task input parameter from UI field , not hard coded in json file. How can we pass the input to task work through an api call.? 

workflow input in json.
{
    "movieId": "234232",
}

"tasks": [
    {
      "name": "movie_info",
      "taskReferenceName": "movie_info",
      "inputParameters": {
        "movieId": "${workflow.input.movieId}"
      },
      "type": "SIMPLE"
    },

I don't want this movieId input to be hardcoded in JSON. Instead it should be from Rest API call as a paramater.

How can we achieve this?

**Comment by manan164** [comment]
Hi @Mishtir , One way to achieve this is to add [HTTP](https://orkes.io/content/docs/reference-docs/system-tasks/http-task) task as the first task and invoke the API there. Now for subsequent tasks we can use http_task.output.resonse.body.status like that.
